compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -1104,6 +1104,23 @@ fn check_type_defn<'tcx>(
         for variant in variants.iter() {
             // All field types must be well-formed.
             for field in &variant.fields {
+                if let Some(def_id) = field.value
```

Comment:
For the record, erroring eagerly even if we don't instantiate the default is consistent with how defaults in struct const generics works:

```rust
struct Foo<const N: u8 = {130 + 130}> {}
```

which errors with:

```shell
error[E0080]: evaluation of constant value failed
 --> src/lib.rs:1:27
  |
1 | struct Foo<const N: u8 = {130 + 130}> {}
  |                           ^^^^^^^^^ attempt to compute `130_u8 + 130_u8`, which would overflow

For more information about this error, try `rustc --explain E0080`.
error: could not compile `playground` (lib) due to 1 previous error
```

Not totally certain why the RFC specified that it would only be linted against, but this seems to me to be the correct behavior.